invenio_records_resources/services/records/queryparser/query.py

```diff
@@ -82,13 +118,18 @@ def parse(self, query_str):
             # We parse the Lucene query syntax in Python, so we know upfront
             # if the syntax is correct before executing it in the search engine
             tree = luqum_parser.parse(query_str)
-
             # Perform transformation on the abstract syntax tree (AST)
-            if self.tree_transformer_factory is not None:
-                transformer = self.tree_transformer_factory()
+            if self.tree_transformer_cls is not None:
+                transformer = self.tree_transformer_cls(
+                    mapping=self.mapping, allow_list=self.allow_list
+                )
                 new_tree = transformer.visit(tree, context={"identity": self.identity})
                 query_str = str(new_tree)
             return dsl.Q("query_string", query=query_str, **self.extra_params)
```

Comment:
question: `extra_params` is passing the list of `fields` to be used in the query. Why do not pass `self.fields` directly given that we are computing them based on the allowed list? 

Same question applies to the multi match query below

Reply:
The `fields` parameter is a bit special:
- If the parameter is given (as is in `extra_params`) it should be passed as is when using query string, to avoid biasing the boosting/filtering. That is why I'm not passing `self.fields`.
- In the case of `multi_field` query the calculated values should only be passed if there is an allowed list. Otherwise, we would overwrite a potential default or set it to `None`. That's why there is an `if` in the except block.